opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1504,6 +1535,27 @@ func generateVm(d *schema.ResourceData, tplContext *dyn.Vector) (string, error)
 }
 
 func resourceVMCustomizeDiff(diff *schema.ResourceDiff, v interface{}) error {
+
+	onChange := diff.Get("on_disk_change").(string)
+	oldDisk, newDisk := diff.GetChange("disk")
+
+	oldDiskList := oldDisk.([]interface{})
+	newDiskList := newDisk.([]interface{})
+
+	for i, oi := range oldDiskList {
+		oiB := oi.(map[string]interface{})
+		niB := newDiskList[i].(map[string]interface{})
+
+		log.Printf("[DEBUG] old disk: %s\n", oiB["image_id"])
+		log.Printf("[DEBUG] new disk: %s\n", niB["image_id"])
+
+		if oiB["image_id"].(int) != niB["image_id"].(int) {
+			if onChange == "recreate" {
```

Comment:
instead of only test this line, we could put this whole part of code in the test (this would save a bit of useless computation in cases onChange != "recreate")

Reply:
Done.